src/clarity-angular/datagrid/datagrid.ts

```diff
@@ -176,18 +186,36 @@ export class Datagrid implements AfterContentInit, AfterViewInit, OnDestroy {
      * by querying the projected content. This is needed to keep track of the models currently
      * displayed, typically for selection.
      */
+
     @ContentChildren(DatagridRow) rows: QueryList<DatagridRow>;
 
+    /**
+     * We get deep nested DatagridActionOverflow children components, listen to the changes in them,
+     * and figure out if the datagrid has at least one actionable row.
+     */
+    @ContentChildren(DatagridActionOverflow, { descendants: true }) public actionableRows: QueryList<DatagridRow>;
+
     ngAfterContentInit() {
-        // TODO: Move all this to ngOnInit() once https://github.com/angular/angular/issues/12818 goes in.
-        // And when we do that, remove the manual step for each one.
+
+        this.actionableRows.changes.subscribe(() => {
```

Comment:
Why did you remove this comment, it should stay there.

Also, it's incredibly relevant here since you forgot to write the explicitly mentioned "manual step". You should do `this.rowActionService.hasActionableRow = this.actionableRows.length > 0;` once on initialization outside of the subscription too, because at that point the QueryList is already populated with the initial value. (See the same behavior on `this.items` right underneath.)

Finally, please add this subscription to `this._subscriptions`, to make sure it's properly unsubscribed on destroy. Once again, see the existing code right after this.